plugins/utils/todo-comments.nix

```
@@ -372,37 +366,44 @@ lib.nixvim.neovim-plugin.mkNeovimPlugin config {
 
   extraOptions = {
     keymaps =
-      let
-        mkKeymapOption =
-          optionName: funcName:
-          mkCompositeOption "Keymap settings for the `:${funcName}` function." {
-            key = mkOption {
-              type = types.str;
-              default = null;
-              description = "Key for the `${funcName}` function.";
-            };
-
-            cwd = mkOption {
-              type = types.nullOr types.str;
-              description = "Specify the directory to search for comments";
-              default = null;
-              example = "~/projects/foobar";
-            };
-
-            keywords = mkOption {
-              type = with types; transitionType str (splitString ",") (nullOr (listOf str));
-              description = ''
-                Comma separated list of keywords to filter results by.
-                Keywords are case-sensitive.
-              '';
-              default = null;
-              example = "TODO,FIX";
+      mapAttrs
+        (
+          optionName: action:
+          mkNullOrOption' {
+            type = keymaps.mkMapOptionSubmodule {
+              defaults = {
+                inherit action;
+                mode = "n";
+              };
+
+              extraOptions = {
+                cwd = mkOption {
+                  type = types.nullOr types.str;
+                  description = "Specify the directory to search for comments";
+                  default = null;
+                  example = "~/projects/foobar";
+                };
+
+                keywords = mkOption {
+                  type = with types; transitionType str (splitString ",") (nullOr (listOf str));
+                  description = ''
+                    Comma separated list of keywords to filter results by.
+                    Keywords are case-sensitive.
+                  '';
+                  default = null;
+                  example = "TODO,FIX";
+                };
+              };
             };
-
-            options = keymaps.mapConfigOptions;
-          };
-      in
-      mapAttrs mkKeymapOption keymapsActions;
+            description = "Keymap for function ${action}";
+          }
+        )
+        {
+          todoQuickFix = "<cmd>TodoQuickFix<cr>";
+          todoLocList = "<cmd>TodoLocList<cr>";
+          todoTrouble = "<cmd>TodoTrouble<cr>";
+          todoTelescope = "<cmd>TodoTelescope<cr>";
```

```suggestion
          todoQuickFix = "TodoQuickFix";
          todoLocList = "TodoLocList";
          todoTrouble = "TodoTrouble";
          todoTelescope = "TodoTelescope";
```

We will add the `<cmd>` and `<cr>` _later_ when defining the top-level keymap's action.

---

Sorry, I misinterpreted a different suggested change